engine/graphics/src/vulkan/graphics_vulkan.cpp

```diff
@@ -2135,23 +2131,35 @@ namespace dmGraphics
             shader->m_UniformBufferCount     = uniform_buffer_count;
         }
 
-        if (ddf->m_Attributes.m_Count > 0)
+        if (ddf->m_Inputs.m_Count > 0)
         {
-            shader->m_Attributes     = new ShaderResourceBinding[ddf->m_Attributes.m_Count];
-            shader->m_AttributeCount = ddf->m_Attributes.m_Count;
-            int32_t  last_binding    = -1;
+            shader->m_Inputs     = new ShaderResourceBinding[ddf->m_Inputs.m_Count];
+            shader->m_InputCount = ddf->m_Inputs.m_Count;
 
-            for (uint32_t i=0; i < ddf->m_Attributes.m_Count; i++)
+            for (uint32_t i=0; i < ddf->m_Inputs.m_Count; i++)
             {
-                ShaderResourceBinding& res = shader->m_Attributes[i];
-                res.m_Binding              = ddf->m_Attributes[i].m_Binding;
-                res.m_Set                  = ddf->m_Attributes[i].m_Set;
-                res.m_Type                 = ddf->m_Attributes[i].m_Type;
-                res.m_Name                 = strdup(ddf->m_Attributes[i].m_Name);
-                res.m_NameHash             = dmHashString64(res.m_Name);
-
-                assert(res.m_Binding >= last_binding);
-                last_binding = res.m_Binding;
+                ShaderResourceBinding& res = shader->m_Inputs[i];
+                res.m_Binding              = ddf->m_Inputs[i].m_Binding;
+                res.m_Set                  = ddf->m_Inputs[i].m_Set;
+                res.m_Type                 = ddf->m_Inputs[i].m_Type;
+                res.m_NameHash             = ddf->m_Inputs[i].m_NameHash;
+                res.m_Name                 = strdup(ddf->m_Inputs[i].m_Name);
+            }
+        }
+
+        if (ddf->m_Outputs.m_Count > 0)
+        {
+            shader->m_Outputs     = new ShaderResourceBinding[ddf->m_Outputs.m_Count];
+            shader->m_OutputCount = ddf->m_Outputs.m_Count;
+
+            for (uint32_t i=0; i < ddf->m_Outputs.m_Count; i++)
+            {
+                ShaderResourceBinding& res = shader->m_Outputs[i];
+                res.m_Binding              = ddf->m_Outputs[i].m_Binding;
+                res.m_Set                  = ddf->m_Outputs[i].m_Set;
+                res.m_Type                 = ddf->m_Outputs[i].m_Type;
+                res.m_NameHash             = ddf->m_Outputs[i].m_NameHash;
+                res.m_Name                 = strdup(ddf->m_Outputs[i].m_Name);
```

Comment:
Not sure if it's worth it but these two segments could be put into a function. it's an edge case I think.